server/lib/nl/config_builder/answer.py

```diff
@@ -17,19 +17,32 @@
 from server.lib.nl.config_builder import base
 from server.lib.nl.fulfillment.types import ChartSpec
 
-_SINGULAR_TITLE = 'The {property} for {entity} is:'
+_ARROW_TO_DIRECTION = {
+    '->': PropertySpec.PropertyDirection.OUT,
+    '<-': PropertySpec.PropertyDirection.IN
+}
+_PROP_ARROW_LENGTH = 2
+_OUT_TITLE = 'The {property} for {entity} is:'
+_IN_TITLE = '{entity} is the {property} for:'
+
+
+def _get_prop_and_direction(
+    prop_str: str) -> tuple[str, PropertySpec.PropertyDirection]:
+  arrow = prop_str[:_PROP_ARROW_LENGTH]
+  prop = prop_str[_PROP_ARROW_LENGTH:]
+  return prop, _ARROW_TO_DIRECTION.get(arrow)
 
 
 def answer_message_block(builder: base.Builder, cspec: ChartSpec):
-  prop = cspec.props[0]
   entity = cspec.entities[0]
-  title = _SINGULAR_TITLE.format(property=prop,
-                                 entity=entity.name or entity.dcid)
+  prop, direction = _get_prop_and_direction(cspec.props[0])
+  title_format_str = _OUT_TITLE if direction == PropertySpec.PropertyDirection.OUT else _IN_TITLE
```

Comment:
nit: maybe a helper to take entity, prop, direction and return a title?

Reply:
done